src/app/tasks/services/tasks-filters.service.spec.ts

```diff
@@ -0,0 +1,150 @@
+import { TaskStatus } from '@aneoconsultingfr/armonik.api.angular';
+import { TestBed } from '@angular/core/testing';
+import { FiltersAnd } from '@app/types/filters';
+import { DefaultConfigService } from '@services/default-config.service';
+import { TableService } from '@services/table.service';
+import { TasksFiltersService } from './tasks-filters.service';
+import { TasksStatusesService } from './tasks-status.service';
+import { TaskFilterDefinition, TaskSummaryFiltersOr } from '../types';
+
+
+describe('TasksFilterService', () => {
+  let service: TasksFiltersService;
+  const mockDefaultConfigService = {
+    defaultTasks: {
+      filters: [] as unknown as TaskSummaryFiltersOr 
+    }
+  };
+  const mockTasksStatusesService = {
+    statuses: {
+      0: $localize`Unspecified`,
+      3: $localize`Dispatched`,
+      1: $localize`Creating`,
+      2: $localize`Submitted`,
+      9: $localize`Processing`,
+      10: $localize`Processed`,
+      7: $localize`Cancelling`,
+      8: $localize`Cancelled`,
+      4: $localize`Finished`,
+      5: $localize`Error`,
+      6: $localize`Timeout`,
+      11: $localize`Retried`,
+    }
+  };
+  const mockTableService = {
+    saveFilters: jest.fn(),
+    resetFilters: jest.fn(),
+    restoreFilters: jest.fn(),
+  };
+  const mockFiltersDefinitions : TaskFilterDefinition[] = [
```

Comment:
The way you are using it is not really mocking. You are just checking that it is the expected result in the 'retrieveDefinitions` test.

A better name would be:
```suggestion
  const expectedFiltersDefinitions: TaskFilterDefinition[] = [
```